src/node_http2_core-inl.h

```diff
@@ -22,6 +22,19 @@ inline int Nghttp2Session::OnNghttpError(nghttp2_session* session,
 }
 #endif
 
+inline bool Nghttp2Session::CanAddNewStream() {
+  uint32_t maxConcurrentStreams =
+      nghttp2_session_get_local_settings(
+          session_,
+          NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS);
+  uint32_t maxSize =
+      std::min(static_cast<uint32_t>(streams_.max_size()),
+               maxConcurrentStreams);
```

Comment:
If this works it’s going to be implementation-dependent luck, wrapping the `size_t` to an `uint32_t` is going to truncate the important bits away on 64-bit platforms. I think upcasting `maxConcurrentStreams` to `size_t` is what you’d rather want here, especially since you’re comparing `streams_.size()` with the result, which generally is a `size_t` too

Also, common core style is `snake_case` for local variables

Reply:
Yeah, good point.